src/services/flowsheet.service.ts

```diff
@@ -164,26 +171,29 @@ const createJoinNotification = async (id: number): Promise<FSEntry> => {
 export const endShow = async (currentShow: Show): Promise<Show> => {
   //Add leave notification for all remaining guest djs;
   //Update their active state and set show end time.
-  const query = sql`
-    INSERT INTO ${flowsheet} (message)
-    SELECT ${djs.dj_name} || ' left the show!'
-      FROM ${djs} INNER JOIN ${show_djs}
-        ON ${show_djs.active} = true AND ${djs.id} != ${currentShow.primary_dj_id} AND ${djs.id} = ${show_djs.dj_id};
-
-    INSERT INTO ${flowsheet} (message)
-    SELECT 'END OF SHOW: ' || ${djs.dj_name}.dj_name || ' SIGNED OFF at ' || to_char(current_timestamp AT TIME ZONE 'US/Eastern', 'HH12:MI AM (MM/DD/YY)')
-      FROM ${djs} 
-      WHERE ${djs.id} = ${currentShow.primary_dj_id};
-
-    UPDATE ${show_djs} SET ${show_djs.active} = false 
-      WHERE ${show_djs.active} = true;
-
-    UPDATE ${shows} SET ${shows.end_time} = CURRENT_TIMESTAMP
-      WHERE ${shows.id} = ${currentShow.id};
-  `;
-
-  const res = await db.execute(query);
-  console.log(res);
+
+  const primary_dj_id = currentShow.primary_dj_id;
+  if (!primary_dj_id) throw new Error('Primary DJ not found');
+
+  const remaining_djs = await db.select().from(show_djs).where(and(
+    eq(show_djs.show_id, currentShow.id),
+    eq(show_djs.active, true)
+  ));
+
+  await Promise.all(remaining_djs.map(async (dj) => {
+    await db.update(show_djs).set({ active: false }).where(eq(show_djs.dj_id, dj.dj_id));
+    if (dj.dj_id === primary_dj_id) return;
+    await createLeaveNotification(dj.dj_id);
+  }));
+
+  const dj_information = (await db.select().from(djs).where(eq(djs.id, primary_dj_id)).limit(1))[0];
+
+  await db.insert(flowsheet).values({
+    message: `End of Show: DJ ${dj_information.dj_name} left the set at ${new Date().toLocaleString('en-US', { timeZone: 'America/New_York' })}`
```

Comment:
Is there any way to handle the timezone of the client? If I were to do a remote set from Seattle, would it make sense for me to see my own timezone?

Reply:
I can write that into the frontend, but for flowsheet-as-legal-FCC-document purposes, what do we want written to the backend?

Reply:
That's up to you. Most systems default to UTC.